core/math/color.cpp

```diff
@@ -595,6 +595,9 @@ Color Color::from_ok_hsl(float p_h, float p_s, float p_l, float p_alpha) {
 }
 
 float Color::get_ok_hsl_h() const {
+	if (r == 0.0 && g == 0.0 && b == 0.0) {
```

Comment:
These comparisons should either be `0` or `0.0f`. `0.0` means `double`.

If `Color` is `float`you are therefore up casting `r` to double before the comparison here.
If `Color` is `double` it will work fine with constant `0` or `0.0f`.